uint8 group_number
---
bool was_set
